### PR TITLE
feat: Add a notebook sample for CPR SDK integration modified from Custom

### PR DESCRIPTION
**Pull request by abcdefgs0324**
Predict notebook.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb [hunk 1]
```diff
@@ -0,0 +1,1311 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "9724b00aeead"
+   },
+   "outputs": [],
+   "source": [
+    "BUCKET_NAME = \"[BUCKET]\"  # @param {type:\"string\"}"
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:363]: Let's rename to gs://[BUCKET]. Here and the other M4 notebook.

I created a PR to change for the M2 notebook already (https://github.com/googleapis/python-aiplatform/pull/1045).

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb [hunk 2]
```diff
@@ -0,0 +1,1311 @@
...
+    "1. [Request Handler](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/handler.py)\n",
+    "    * Responsible for webserver aspects of handling a request, such as deserializing the request body, and serializing the reponse, setting response headers, etc.\n",
+    "    * In this example, we will use the default Handler, `google.cloud.aiplatform.prediction.handler.PredictionHandler` provided in the SDK.\n",
+    "1. [Predictor](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/predictor.py)\n",
+    "    * Responsible for the ML logic for processing a prediction request.\n",
+    "\n",
+    "Each of these three pieces can be customized based on the requirements of the custom container. In this example, we will only be modify the `Predictor`.\n",
+    "\n",
+    "\n",
+    "A [`Predictor`](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/predictor.py)must implement the following interface:"
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:582]: Missing a space between `Predictor` and the next word.

Please fix for all 3.

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb [hunk 3]
```diff
@@ -0,0 +1,1311 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "7wj8T6yA5Ud4"
+   },
+   "outputs": [],
+   "source": [
+    "class Predictor:\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:593]: Change this code block to markdown. I've already done so for the M2 notebook in https://github.com/googleapis/python-aiplatform/pull/1045.

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb [hunk 4]
```diff
@@ -0,0 +1,1311 @@
...
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from google.cloud.aiplatform.prediction import LocalModel\n",
+    "\n",
+    "from src_dir.predictor import CprPredictor  # Update this path as the variable $USER_SRC_DIR to import the custom predictor.\n",
+    "\n",
+    "local_model = LocalModel.create_cpr_model(\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:837]: I really love how much simpler the interface is for users :)

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb [hunk 5]
```diff
@@ -0,0 +1,1434 @@
...
+    "A [`Handler`](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/handler.py) must implement the following interface."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Handler:\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb:716]: Change this block to markdown too.

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb [hunk 6]
```diff
@@ -0,0 +1,1434 @@
...
+    "from typing import Optional, Type\n",
+    "\n",
+    "class CprHandler(Handler):\n",
+    "    \"\"\"Default prediction handler for the prediction requests sent to the application.\"\"\"\n",
+    "\n",
+    "    def __init__(\n",
+    "        self, gcs_artifacts_uri, predictor = None,\n",
+    "    ):\n",
+    "        \"\"\"Initializes a Handler instance.\"\"\"\n",
+    "        if predictor is None:\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb:775]: Can we inherit from PredictionHandler, and not implement this part?

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb [hunk 7]
```diff
@@ -0,0 +1,1434 @@
...
+    "                \"CprHandler must have a predictor class passed to the init function.\"\n",
+    "            )\n",
+    "\n",
+    "        self._predictor = predictor()\n",
+    "        self._predictor.load(gcs_artifacts_uri)\n",
+    "\n",
+    "    async def handle(self, request):\n",
+    "        \"\"\"Handles a prediction request.\"\"\"\n",
+    "        request_body = await request.body()\n",
+    "        data = json.loads(request_body)\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb:786]: I was thinking the request would not be json encoded. Instead it'll be raw CSV

curl -X POST -H 'Content-Type: text/csv' -d @file.csv [ENDPOINT]

so we'd csv decode the request body itself.

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb (hunk 1 above)
Comment (abcdefgs0324) [reply samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb]: Done.

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb (hunk 2 above)
Comment (abcdefgs0324) [reply samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb]: Done.

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb (hunk 3 above)
Comment (abcdefgs0324) [reply samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:585]: Done.

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb (hunk 4 above)
Comment (abcdefgs0324) [reply samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:830]: : )

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb (hunk 6 above)
Comment (abcdefgs0324) [reply samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb]: Done.

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb (hunk 5 above)
Comment (abcdefgs0324) [reply samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb:702]: Done.

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb (hunk 7 above)
Comment (abcdefgs0324) [reply samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb]: Done. Also, changed it to raw-predict.

**Review by kthytang — APPROVED** [review]
Awesome work, thanks!

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb [hunk 8]
```diff
@@ -0,0 +1,1304 @@
...
+    "Write the dependencies to the source directory which will be installed in the image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile $USER_SRC_DIR/requirements.txt\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:715]: Can we replace this with 
```
!cp requirements.txt $USER_SRC_DIR/requirements.txt
```
?

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb [hunk 9]
```diff
@@ -0,0 +1,1417 @@
...
+    "Write the dependencies to the source directory which will be installed in the image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile $USER_SRC_DIR/requirements.txt\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb:807]: Same here.

File: samples/notebooks/prediction/SDK_Custom_Predict.ipynb [hunk 10]
```diff
@@ -550,7 +550,7 @@
         "Each of these three pieces can be customized based on the requirements of the custom container. In this example, we will only be modify the `Predictor`.\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict.ipynb:550]: "will only be modify" -> "will only be implementing"

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb [hunk 11]
```diff
@@ -0,0 +1,1304 @@
...
+    "    * HTTP server that hosts the model\n",
+    "    * Responsible for setting up routes/ports/etc.\n",
+    "    * In this example we will use the `google.cloud.aiplatform.prediction.model_server.ModelServer` out of the box.\n",
+    "1. [Request Handler](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/handler.py)\n",
+    "    * Responsible for webserver aspects of handling a request, such as deserializing the request body, and serializing the reponse, setting response headers, etc.\n",
+    "    * In this example, we will use the default Handler, `google.cloud.aiplatform.prediction.handler.PredictionHandler` provided in the SDK.\n",
+    "1. [Predictor](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/predictor.py)\n",
+    "    * Responsible for the ML logic for processing a prediction request.\n",
+    "\n",
+    "Each of these three pieces can be customized based on the requirements of the custom container. In this example, we will only be modify the `Predictor`.\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:579]: "will only be modify" -> "will only be implementing"

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb [hunk 12]
```diff
@@ -0,0 +1,1417 @@
...
+    "    * HTTP server that hosts the model\n",
+    "    * Responsible for setting up routes/ports/etc.\n",
+    "    * In this example we will use the `google.cloud.aiplatform.prediction.model_server.ModelServer` out of the box.\n",
+    "1. [Request Handler](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/handler.py)\n",
+    "    * Responsible for webserver aspects of handling a request, such as deserializing the request body, and serializing the reponse, setting response headers, etc.\n",
+    "    * In this example, we will use the default Handler, `google.cloud.aiplatform.prediction.handler.PredictionHandler` provided in the SDK.\n",
+    "1. [Predictor](https://github.com/googleapis/python-aiplatform/blob/custom-prediction-routine/google/cloud/aiplatform/prediction/predictor.py)\n",
+    "    * Responsible for the ML logic for processing a prediction request.\n",
+    "\n",
+    "Each of these three pieces can be customized based on the requirements of the custom container. In this example, we will modify the `Predictor` and `Handler`.\n",
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb:580]: modify -> implement

**Review by kthytang — APPROVED** [review]

File: samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb [hunk 13]
```diff
@@ -0,0 +1,1304 @@
...
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "-KODNAqK5k8R"
+   },
+   "source": [
+    "First, implement a custom `Predictor` that loads in both the preprocesor and the model. The preprocessor and the model when then be used at `predict` time."
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_SDK_Integration.ipynb:637]: "when then" -> "will then"

Please update in the M2 notebook as well.

File: samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb [hunk 14]
```diff
@@ -0,0 +1,1417 @@
...
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "-KODNAqK5k8R"
+   },
+   "source": [
+    "First, implement a custom `Predictor` that loads in both the preprocesor and the model. The preprocessor and the model when then be used at `predict` time."
```
Comment (kthytang) [inline samples/notebooks/prediction/SDK_Custom_Predict_and_Handler_SDK_Integration.ipynb:638]: "when then" -> "will then"

**Review by kthytang — APPROVED** [review]

**Pull request merged**